br/pkg/lightning/backend/external/stat_reader.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package external
+
+import (
+	"context"
+	"encoding/binary"
+
+	"github.com/pingcap/tidb/br/pkg/storage"
+)
+
+type statsReader struct {
+	byteReader *byteReader
+}
+
+func newStatsReader(ctx context.Context, store storage.ExternalStorage, name string, bufSize int) (*statsReader, error) {
+	sr, err := openStoreReaderAndSeek(ctx, store, name, 0)
+	if err != nil {
+		return nil, err
+	}
+	br, err := newByteReader(ctx, sr, bufSize)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Same with kv_reader.go: 30-34

Reply:
`initFileOffset` is different. I think it is simple enough.